src/normalize.js

```diff
@@ -5,68 +5,97 @@ const extractFields = async (
   store,
   cache,
   createNode,
+  createNodeId,
   touchNode,
   auth,
-  item
+  item,
+  key = 'localFile'
 ) => {
-  for (const key of Object.keys(item)) {
-    const field = item[key]
-    if (Array.isArray(field)) {
-      // add recursion to fetch nested strapi references
-      await Promise.all(
-        field.map(async f =>
-          extractFields(apiURL, store, cache, createNode, touchNode, auth, f)
-        )
-      )
-    } else {
-      // image fields have a mime property among other
-      // maybe should find a better test
-      if (field !== null && field.hasOwnProperty('mime')) {
-        let fileNodeID
-        // using field on the cache key for multiple image field
-        const mediaDataCacheKey = `strapi-media-${item.id}-${key}`
-        const cacheMediaData = await cache.get(mediaDataCacheKey)
+  // image fields have a mime property among other
+  // maybe should find a better test
+  if (item && item.hasOwnProperty('mime')) {
+    let fileNodeID
+    // using field on the cache key for multiple image field
+    const mediaDataCacheKey = `strapi-media-${item.id}-${key}`
+    const cacheMediaData = await cache.get(mediaDataCacheKey)
 
-        // If we have cached media data and it wasn't modified, reuse
-        // previously created file node to not try to redownload
-        if (cacheMediaData && field.updated_at === cacheMediaData.updated_at) {
-          fileNodeID = cacheMediaData.fileNodeID
-          touchNode({ nodeId: cacheMediaData.fileNodeID })
-        }
+    // If we have cached media data and it wasn't modified, reuse
+    // previously created file node to not try to redownload
+    if (cacheMediaData && item.updatedAt === cacheMediaData.updatedAt) {
```

Comment:
Actually, this should be an option as the timestamps are configurable per model. (e.g by default on mongo it is camelcased `createdAt / updatedAt` and on sql databases it would be `created_at / updated_at`.)

Reply:
Oh I didn't know that, I just changed it because at some point in the code the variables was set as updatedAt but then inside that if the value being used was updated_at which was always undefined for and I just though that it was an inconsistency. I can try to test it and fix it tomorrow.

Reply:
@alexandrebodin I fixed it with your suggestion but I have a question.
What I did is to create a new variable like this: `const itemUpdatedAt = item.updatedAt || item.updated_at`, is there a better way to check what database is being used or this method should be fine?

Reply:
Hi, This should be fine for now. We will need to fetch model metadata to be more dynamic in the future though :) Will test this weekend  se we can release next week ;)